lightning/src/ln/wire.rs

```diff
@@ -238,6 +263,18 @@ pub trait Encode {
 	}
 }
 
+/// A message that has an associated type id.
+pub trait TypedMessage {
+	/// The type id for the implementing message.
+	fn msg_type(&self) -> u16;
```

Comment:
Does it make sense to use the `MessageType` newtype here (and basically in all the new code with message types) instead of `u16`?

Reply:
You mean the `MessageType(u16)`? I tried at some point but it would require making the u16 public, and then from a user perspective you'd have to wrap your `u16` into `MessageType` and unwrap them when matching, so I didn't feel like it provided big benefit.
If I'm being honest even within LDK I didn't feel like it provided a lot apart from the `is_even` method, but maybe I'm missing something. It would be nice if it could somehow abstract the underlying type but since it needs to be serialized as a u16 at the end I'm not sure it's possible.

Also even with the nice dummy wrapper module it doesn't seem to be possible to make `MessgeType` non public.

Reply:
I don't feel strongly, tbh, it just felt awkward to have a newtype and not use it, maybe @jkczyz feels strongly.

Reply:
Sorry I fell off the review. I think we could get rid of `MessageType` and clean up `Encode` by moving `type_id` to the new trait. Do you mind taking the commits in https://github.com/jkczyz/rust-lightning/tree/2011-08-wire-type? I did this there along with some other clean-ups with naming. It moves `is_even` to `Message`, which can be used on the `Custom` variant.

Reply:
I cherry picked your commits. It makes things really clean IMHO, thanks!